Controller/ConfigController.php

```diff
@@ -141,6 +155,41 @@ public function editAction(Request $request, $id)
             throw $this->createNotFoundException('Not Found', $e);
         }
 
+        if($data->getAdapter() == "twitter" && $request->getMethod() !== "PUT" && empty($data->getOptions()->get("token")) && empty($data->getOptions()->get("token_secret")))
```

Comment:
Can you think about the code; which makes it more dynamically so new adapters can be added easily instead of copying and changing some values?